test-data/unit/check-varargs.test

```diff
@@ -1066,3 +1117,166 @@ class C:
+
+def src(**kwargs: Unpack[Vehicles]): ...
+
+dest = src  # E: Incompatible types in assignment (expression has type "Callable[[KwArg(Vehicles)], Any]", variable has type "Callable[[KwArg(Vehicle)], Any]")
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsAssignmentWhenOnlySourceContainsKwargs]
+from typing import NotRequired
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+class Example(TypedDict):
+    animal: Animal
+    string: str
+    number: NotRequired[int]
+
+def src(**kwargs: Unpack[Example]): ...
+def dest(*, animal: Dog, string: str, number: int = ...): ...
+
+dest = src  # OK
+
+def dest_with_positional_args(animal: Dog, string: str, number: int = ...): ...
+
+dest_with_positional_args = src  # E: Incompatible types in assignment (expression has type "Callable[[KwArg(Example)], Any]", variable has type "Callable[[Dog, str, int], Any]")
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testUnpackKwargsAssignmentWhenOnlyDestinationContainsKwargs]
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+def dest(**kwargs: Unpack[Animal]): ...
+def src(name: str): ...
+
+dog: Dog = {"name": "Daisy", "breed": "Labrador"}
+animal: Animal = dog
+
+dest = src  # E: Incompatible function signatures - variable contains **kwargs and the expression does not. See PEP 692 for more details.
+
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testUnpackKwargsAssignmentWhenBothSidesContainKwargsTypedWithUnpack]
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+def accept_animal(**kwargs: Unpack[Animal]): ...
+def accept_dog(**kwargs: Unpack[Dog]): ...
+
+accept_dog = accept_animal  # OK
+
+def also_accept_animal(**kwargs: Unpack[Animal]): ...
+def also_accept_dog(**kwargs: Unpack[Dog]): ...
+
+also_accept_animal = also_accept_dog  # E: Incompatible types in assignment (expression has type "Callable[[KwArg(Dog)], Any]", variable has type "Callable[[KwArg(Animal)], Any]")
+
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testPassingKwargsFromFunctionToFunction]
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+def takes_name(name: str):
+    ...
+
+dog: Dog = {"name": "Daisy", "breed": "Labrador"}
+animal: Animal = dog
+
+def foo(**kwargs: Unpack[Animal]):
+    return kwargs["name"]
+
+def bar(**kwargs: Unpack[Animal]):
+    takes_name(**kwargs)
+
+def baz(animal: Animal):
+    takes_name(**animal)
+
+def spam(**kwargs: Unpack[Animal]):
+    baz(kwargs)
+
+foo(**animal)  # OK! foo only expects and uses keywords of 'Animal'.
+
+bar(**animal)  # E: This will fail at runtime because 'breed' keyword will be passed to 'takes_name' as well.
```

Comment:
Trying to support this is futile (not just in mypy, in any reasonable type checker), since `foo` and `bar` have identical signatures, you can't make one fail and other pass for an identical call.

Reply:
Hmm, makes sense - probably another thing that PEP has to be amended for.

Reply:
Would it make sense in your opinion to report an error here instead:
```
def bar(**kwargs: Unpack[Animal]):
    takes_name(**kwargs) # E: Potentially passing more arguments than expected, because a subtype of Animal could have been passed
```
?

Reply:
Oh, nevermind, saw the comment below, so the answer probably would be "no" 😅

Reply:
This will break a ton of existing, unless you will make this check apply only when `x` in `**x` comes from `Unpack` in a function, but even then it will likely break some code. This feels like sacrificing a useful feature in favor of something no one asked for.